graspologic/cluster/kclust.py

```diff
@@ -9,6 +11,8 @@
 
 
 class KMeansCluster(BaseCluster):
+    ari_: Optional[List[float]]
```

Comment:
curious why this is necessary for this particular attribute

Reply:
When MyPy first encounters `self.ari_` in the `fit` method, it eagerly types it as a List. However, in the else clause in the block where it's assigned it can also take the value `None`:

https://github.com/microsoft/graspologic/blob/3a0be76aea80bed7be16cb8bdb49df8a890ef7ee/graspologic/cluster/kclust.py#L111-L120

So if you don't include the attribute type hint, MyPy gets confused:

```
graspologic\cluster\kclust.py:117: error: Incompatible types in assignment (expression has type "None", variable has type "List[Any]")
Found 1 error in 1 file (checked 79 source files)
```

There were a few spots where it was neccessary to make this kind of forward declaration. An alternative approach would be to declare a local var with type hinting and then assign to `self.ari_` after the if/else block. It seemed like a toss up as to which approach is better.

Reply:
my only concern was stuff like this https://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.check_is_fitted.html which could end up checking for whether that attribute exists or not. I don't feel too strongly, though.

Reply:
Type hints without an assignment expression don't actually create the attribute:

```python3
class C:
    some_attr_: int

    def __init__(self):
        self.attr_from_init_ = 42

    def f(self):
        self.some_attr_ = 11

c = C()
print("after init", c.__dict__)
c.f()
print("after f", c.__dict__)
```

outputs

```
after init {'attr_from_init_': 42}
after f {'attr_from_init_': 42, 'some_attr_': 11}
```

Reply:
oh, great i just misunderstood then. thanks!